lib/core/src/restore.rs

```diff
@@ -0,0 +1,820 @@
+
+                    // If a history tx is a known incoming txs, it's the claim tx
+                    let (lockup_tx_id, claim_tx_id) =
+                        match tx_map.incoming_tx_map.contains_key::<Txid>(first) {
+                            true => (second, first),
+                            false => (first, second),
+                        };
+                    lbtc_server_lockup_claim_tx_ids
+                        .insert(swap_id.clone(), (*lockup_tx_id, *claim_tx_id));
+                }
+                n => {
+                    error!("L-BTC script history with unexpected length {n} found while recovering data for Chain Receive Swap {swap_id}")
+                }
+            }
+
+            match history.btc_lockup_script_history.len() {
+                // TODO How to treat case when history length > 2? (address re-use)
+                x if x >= 2 => {
+                    // TODO How to tell the user lockup tx apart from the refund tx? Is the order in which they're received from Electrum reliable?
+                    // TODO How to tell BTC refund apart from BTC server claim tx?
+                    btc_user_lockup_tx_ids
+                        .insert(swap_id.clone(), history.btc_lockup_script_history[0]);
+                    btc_refund_tx_ids.insert(swap_id.clone(), history.btc_lockup_script_history[1]);
+                }
+                n => {
+                    error!("BTC script history with unexpected length {n} found while recovering data for Chain Receive Swap {swap_id}")
+                }
+            }
+        }
+
+        info!("[Recover Chain Receive] Found {} L-BTC server lockup and claim tx pairs from onchain data", lbtc_server_lockup_claim_tx_ids.len());
+        info!(
+            "[Recover Chain Receive] Found {} BTC user lockup tx IDs from onchain data",
+            btc_user_lockup_tx_ids.len()
+        );
+        info!(
+            "[Recover Chain Receive] Found {} BTC refund tx IDs from onchain data",
+            btc_refund_tx_ids.len()
+        );
+
+        Ok(RecoveredOnchainDataChainReceive {
+            lbtc_server_lockup_claim_tx_ids,
+            btc_user_lockup_tx_ids,
+            btc_refund_tx_ids,
+        })
+    }
+}
+
+/// Methods to simulate the immutable DB data available from real-time sync
+// TODO Remove once real-time sync is integrated
+pub(crate) mod immutable {
+    use std::collections::HashMap;
+
+    use anyhow::{anyhow, ensure, Result};
+    use boltz_client::{BtcSwapScript, LBtcSwapScript};
+    use log::{error, info};
+    use lwk_wollet::elements::Txid;
+
+    use crate::prelude::*;
+    use crate::sdk::LiquidSdk;
+
+    type BtcScript = lwk_wollet::bitcoin::ScriptBuf;
+    type LBtcScript = lwk_wollet::elements::Script;
+
+    pub(crate) type SendSwapHistory = Vec<Txid>;
+    pub(crate) type ReceiveSwapHistory = Vec<Txid>;
+
+    #[allow(dead_code)]
+    #[derive(Clone)]
+    struct SendSwapImmutableData {
```

Comment:
NIT: also here Immutable seems too generic. Basically all we have here is the swap_script (event he script is derive from it). So perhapas we can rename to SendSwapScriptsData ?

Reply:
I originally started with that, but later realized these structs are inputs from the Immutable DB, representing the immutable data for the swaps. My understanding is that immutable data will be read into such structs, bringing everything that's available (create response, keypairs, etc). Restore is then supposed to use them as inputs.

From this struct, it happens I'm only using the script data for restoring. I named it `SendSwapImmutableData` in the expectation it will include the rest of the immutable data as well.